src/main/java/org/truffleruby/core/rope/RopeOperations.java

```diff
@@ -111,6 +111,23 @@ public static LeafRope create(byte b, Encoding encoding, CodeRange codeRange) {
         return create(new byte[] { b }, encoding, codeRange);
     }
 
+    public static Rope emptyRope(Encoding encoding) {
+        if (encoding == UTF8Encoding.INSTANCE) {
+            return RopeConstants.EMPTY_UTF8_ROPE;
+        }
+
+        if (encoding == USASCIIEncoding.INSTANCE) {
+            return RopeConstants.EMPTY_US_ASCII_ROPE;
+        }
+
+        if (encoding == ASCIIEncoding.INSTANCE) {
+            return RopeConstants.EMPTY_ASCII_8BIT_ROPE;
+        }
+
+        final CodeRange codeRange = encoding.isAsciiCompatible() ? CR_7BIT : CR_VALID;
+        return withEncodingVerySlow(RopeConstants.EMPTY_ASCII_8BIT_ROPE, encoding, codeRange);
```

Comment:
Maybe create() with a `byte[]` would be simpler here?
And the `byte[]` could be once allocated and shared with other empty ropes since it has length 0.

Reply:
Possibly. The idea here is we already have a shared byte[] and the encoding, but the code range needs to be determined.